### PR TITLE
Integrating Hardhat with Web3.js

### PR DESCRIPTION
**Pull request by Thanos107**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/CONTRIBUTIONS.md).

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: docs/docs/guides/hardhat-guide/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        emit Withdrawal(address(this).balance, block.timestamp);
+
+        owner.transfer(address(this).balance);
+    }
+}
+
+```
+
+- Now let's look at the Javascript File named `deploy.js` from the folder `scripts` and use web3.js to deploy the contract onto the `Blockchain`  using `Hardhat`
```
Comment (jdevcs) [inline docs/docs/guides/hardhat-guide/index.md:116]: @Thanos107 Thanks for your contributions. Its not complete yet.

Comment (jdevcs) [reply docs/docs/guides/hardhat-guide/index.md:116]: are you still working in this PR?

File: docs/package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
 		"@mdx-js/react": "^3.0.0",
 		"classnames": "^2.3.2",
 		"clsx": "^2.0.0",
+		"docusaurus": "^1.14.7",
```
Comment (Muhammad-Altabba) [inline docs/package.json:24]: This is a deprecated version. The new versions start with `@docusaurus/[sub-package-name]` (https://www.npmjs.com/search?q=%40docusaurus). And we had the needed sub-packages already added.
```suggestion
```

**Review by Muhammad-Altabba — COMMENTED** [review]
Thanks @Thanos107,
Are you facing issues with testing your edit? You can check https://github.com/web3/web3.js/tree/4.x/docs#readme for how to install, build and run the documentation website at your machine.
I added a comment for the added `docusaurus` dependency.
Let us know if you wish to continue writing this document so we can assist you.
Thanks,

File: docs/docs/guides/hardhat-guide/index.md (hunk 1 above)
Comment (Thanos107) [reply docs/docs/guides/hardhat-guide/index.md:116]: Yes I am.

**Pull request closed without merging**